src/webui/xmltv.c

```diff
@@ -222,6 +223,48 @@ http_xmltv_programme_one_long(const http_connection_t *hc,
     }
     htsbuf_append_str(hq, "  </credits>\n");
   }
+
+  if (epggrab_conf.epgdb_processparentallabels && ebc->rating_label) {
+      const char *system = ebc->rating_label->rl_authority;
+      if (!system) system = ebc->rating_label->rl_country;
+      if (system) {
+          htsbuf_append_str(hq, "  <rating system=\"");
+          htsbuf_append_and_escape_xml(hq, system);
+          htsbuf_append_str(hq, "\">\n");
+          if (ebc->rating_label->rl_display_label) {
+              htsbuf_append_str(hq, "    <value>");
+              htsbuf_append_and_escape_xml(hq, ebc->rating_label->rl_display_label);
+              htsbuf_append_str(hq, "</value>\n");
+          } else if (ebc->rating_label->rl_display_age) {
+              htsbuf_append_str(hq, "    <value>");
+              htsbuf_qprintf(hq, "%d", ebc->rating_label->rl_display_age);
+              htsbuf_append_str(hq, "</value>\n");
+          } else if (ebc->rating_label->rl_age) {
+              htsbuf_append_str(hq, "    <value>");
+              htsbuf_qprintf(hq, "%d", ebc->rating_label->rl_age);
+              htsbuf_append_str(hq, "</value>\n");
+          }
+          if (ebc->rating_label->rl_icon) {
+              htsbuf_append_str(hq, "    <icon src=\"");
+              htsbuf_append_and_escape_xml(hq, ebc->rating_label->rl_icon);
+              htsbuf_append_str(hq, "\"/>\n");
+          }
+          htsbuf_append_str(hq, "  </rating>\n");
+      } else if (ebc->age_rating) {
+        htsbuf_append_str(hq, "  <rating system=\"AGE\">\n");
+        htsbuf_append_str(hq, "    <value>");
+        htsbuf_qprintf(hq, "%d", ebc->age_rating);
+        htsbuf_append_str(hq, "</value>\n");
+        htsbuf_append_str(hq, "  </rating>\n");
+      }
+  } else if (ebc->age_rating) {
```

Comment:
Looks like this is the same as the code block above this? If yes, please just add it outside the if-block. Or maybe this is not intended?